src/toil/common.py

```diff
@@ -796,13 +796,15 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         """
         Clean up after a workflow invocation. Depending on the configuration, delete the job store.
         """
-        try:
+        try: 
             if (exc_type is not None and self.config.clean == "onError" or
                             exc_type is None and self.config.clean == "onSuccess" or
                         self.config.clean == "always"):
+
                 try:
-                    self._jobStore.destroy()
-                    logger.info("Successfully deleted the job store: %s" % str(self._jobStore))
+                    with self._jobStore.readSharedFileStream('succeeded.log') as successful:                 
```

Comment:
What you want is to check for status here, but the problem is, the 'succeeded.log' file is only present when you're in the SUCCESS state.  You want to be able to get the status back (without erroring!) for any state.

I would use the full getStatus function here: https://github.com/DataBiosphere/toil/blob/2d4e0cdc188a6c0d4c21a6a968bf80ec6b0473bc/src/toil/utils/toilStatus.py#L191

Notice the error is consumed when 'succeeded.log' does not exist.  This should work for all jobstores (AWS/Google/Locally).